### PR TITLE
project structure

### LINKED ISSUE #1: Setup project structure
setup project with react native in `src/mobile`

### PR DESCRIPTION
**Pull request by toxdes**
For #1 Created a basic `Hello world` project, with typescript template. Suggestions are welcome.

### CODE REVIEW

File: src/mobile/package.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    "@react-native-community/eslint-config": "^2.0.0",
+    "@types/jest": "^26.0.23",
+    "@types/react-native": "^0.64.5",
+    "@types/react-test-renderer": "^16.9.2",
+    "babel-jest": "^26.6.3",
+    "eslint": "^7.14.0",
+    "jest": "^26.6.3",
+    "metro-react-native-babel-preset": "^0.64.0",
+    "react-test-renderer": "17.0.1",
+    "typescript": "^3.8.3"
```
Comment (ankitbko) [inline src/mobile/package.json:28]: Can we use latest version?

File: src/mobile/app.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "mobile",
```
Comment (ankitbko) [inline src/mobile/app.json:2]: Do we want to change the name?

File: src/mobile/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  "compilerOptions": {
+    /* Basic Options */
+    "target": "esnext",                       /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+    "lib": ["es2017"],                        /* Specify library files to be included in the compilation. */
+    "allowJs": true,                          /* Allow javascript files to be compiled. */
+    // "checkJs": true,                       /* Report errors in .js files. */
+    "jsx": "react-native",                    /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
+    // "declaration": true,                   /* Generates corresponding '.d.ts' file. */
+    // "sourceMap": true,                     /* Generates corresponding '.map' file. */
```
Comment (ankitbko) [inline src/mobile/tsconfig.json:12]: enable this

File: src/mobile/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    // "removeComments": true,                /* Do not emit comments to output. */
+    "noEmit": true,                           /* Do not emit outputs. */
+    // "incremental": true,                   /* Enable incremental compilation */
+    // "importHelpers": true,                 /* Import emit helpers from 'tslib'. */
+    // "downlevelIteration": true,            /* Provide full support for iterables in 'for-of', spread, and destructuring when targeting 'ES5' or 'ES3'. */
+    "isolatedModules": true,                  /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */
+
+    /* Strict Type-Checking Options */
+    "strict": true,                           /* Enable all strict type-checking options. */
+    // "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
```
Comment (ankitbko) [inline src/mobile/tsconfig.json:25]: and this

File: src/mobile/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    // "strictNullChecks": true,              /* Enable strict null checks. */
+    // "strictFunctionTypes": true,           /* Enable strict checking of function types. */
+    // "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
+    // "noImplicitThis": true,                /* Raise error on 'this' expressions with an implied 'any' type. */
+    // "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */
+
+    /* Additional Checks */
```
Comment (ankitbko) [inline src/mobile/tsconfig.json:32]: and all these

File: src/mobile/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    // "preserveSymlinks": true,              /* Do not resolve the real path of symlinks. */
+    "skipLibCheck": false                      /* Skip type checking of declaration files. */
+
+    /* Source Map Options */
+    // "sourceRoot": "./",                    /* Specify the location where debugger should locate TypeScript files instead of source locations. */
+    // "mapRoot": "./",                       /* Specify the location where debugger should locate map files instead of generated locations. */
+    // "inlineSourceMap": true,               /* Emit a single file with source maps instead of having a separate file. */
+    // "inlineSources": true,                 /* Emit the source alongside the sourcemaps within a single file; requires '--inlineSourceMap' or '--sourceMap' to be set. */
+
+    /* Experimental Options */
```
Comment (ankitbko) [inline src/mobile/tsconfig.json:56]: and these

File: src/mobile/.prettierrc.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  bracketSpacing: false,
+  jsxBracketSameLine: true,
+  singleQuote: true,
+  trailingComma: 'all',
+  arrowParens: 'avoid',
+};
```
Comment (ankitbko) [inline src/mobile/.prettierrc.js:7]: add following
```
"printWidth": 128,

```

**Review by ankitbko — CHANGES_REQUESTED** [review]

File: src/mobile/tsconfig.json (hunk 6 above)
Comment (Cinnybun) [reply src/mobile/tsconfig.json:56]: Should I click Resolve converstation?

**Comment by ankitbko** [comment]
Please add a `setup.md` and document list of softwares (android sdk, android platform version, node ver etc) which are necessary to build this app. And probably also include instructions on how to build/test/run the app.

**Review by Cinnybun — DISMISSED** [review]
LGTM

File: src/mobile/setup.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# Development
```
Comment (ankitbko) [inline src/mobile/setup.md:1]: Also add the prerequisites - 
1. Android SDK
2. JAVA

Also add how to generate avd for android emulator after installing SDK -

```
sdkmanager --install 'system-images;android-29;google_apis_playstore;x86_64'
sdkmanager --install 'extras;intel;Hardware_Accelerated_Execution_Manager'

.\avdmanager.bat create avd -f -n crimsonAVD -d 17 -b google_apis/x86_64 --package 'system-images;android-29;google_apis;x86_64'
```

Comment (ankitbko) [reply src/mobile/setup.md:1]: And how to run the emulator - 

```
emulator -avd crimsonAVD
```

**Review by ankitbko — DISMISSED** [review]

**Comment by tarunsingh495** [comment]
Tested locally. Works fine for me.

**Review by ME-ON1 — APPROVED** [review]
looks fine to me

**Review by tarunsingh495 — APPROVED** [review]
tested Locally . Works fine for me.

**Review by darkfusion90 — APPROVED** [review]

**Pull request merged**